File: backend/PATH.Domain/Entities/Organization.cs
using PATH.Domain.Models;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace PATH.Domain.Entities
{
    public class Organization
    {
        public Guid Id { get; set; }
        public string Name { get; set; }

        public ApplicationUser CreatedBy { get; set; }

        [ForeignKey(nameof(CreatedBy))]
        public Guid CreatedById { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public ICollection<OrganizationMember> Members { get; set; } = new HashSet<OrganizationMember>();

        public ICollection<Project> Projects { get; set; } = new HashSet<Project>();


    }
}

File: frontend/PATH.Web/src/app/features/organization/models/Organization.ts
import { OrgMember } from './OrgMember';
import { Project } from '../../project/models/Project';
import { ProjectMember } from '../../project/models/ProjectMember';

export interface Organization {
  id: string;
  name: string;
  createdById: string;
  createdAt: string;
  creatorName: string;
  currentUserRole: string;
  projects: Project[];
  members: OrgMember[];
}
